x-pack/legacy/plugins/lens/public/indexpattern_plugin/indexpattern_suggestions.tsx

```diff
@@ -298,17 +298,73 @@ function createNewLayerWithMetricAggregation(
 export function getDatasourceSuggestionsFromCurrentState(
   state: IndexPatternPrivateState
 ): Array<DatasourceSuggestion<IndexPatternPrivateState>> {
-  const layers = Object.entries(state.layers);
+  const layers = Object.entries(state.layers || {});
 
-  return layers
-    .map(([layerId, layer], index) => {
+  return _.flatten(
+    layers.map(([layerId, layer], index) => {
       if (layer.columnOrder.length === 0) {
-        return;
+        return [];
       }
 
-      return buildSuggestion({ state, layerId, isMultiRow: true, datasourceSuggestionId: index });
+      const onlyMetric = layer.columnOrder.every(columnId => !layer.columns[columnId].isBucketed);
+      const onlyBucket = layer.columnOrder.every(columnId => layer.columns[columnId].isBucketed);
+      if (onlyMetric || onlyBucket) {
+        // intermediary chart, don't try to suggest reduced versions
+        return [
```

Comment:
No need to return an array, since flatten handles raw values and arrays interchangeably.